ci-operator/step-registry/ipi/install/heterogeneous/ipi-install-heterogeneous-commands.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+dir=/tmp/installer
+mkdir "${dir}/"
+
+# Make sure yq is installed
+if [ ! -f /tmp/yq-v4 ]; then
+  # TODO move to image
+  curl -L "https://github.com/mikefarah/yq/releases/download/v4.25.3/yq_linux_$(uname -m | sed s/aarch64/arm64/ | sed s/x86_64/amd64/)" -o /tmp/yq-v4 && chmod +x /tmp/yq-v4
+  PATH=${PATH}:/tmp
+fi
+
+# Make sure jq is installed
+if ! command -v jq; then
+  # TODO move to image
+  curl -sL https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 > /tmp/jq
+  chmod +x /tmp/jq
+  PATH=${PATH}:/tmp
+fi
+
+REGION=${LEASED_RESOURCE}
+
+echo "Creating arm worker machineset"
+oc -n openshift-machine-api get -o json machinesets | jq '[.items[] | select(.spec.template.metadata.labels["machine.openshift.io/cluster-api-machine-role"] == "worker")][0]' > ${dir}/openshift/99_openshift-cluster-api_worker-machineset-0.yaml
+
+# AMI for AWS ARM
+amiid_workers_additional=$(openshift-install coreos print-stream-json | yq-v4 ".architectures.${ADDITIONAL_WORKER_ARCHITECTURE}.images.aws.regions.\"${REGION}\".image")
+yq-v4 eval ".metadata.name += \"-additional\"
+| .spec.replicas = ${ADDITIONAL_WORKERS}
+| .spec.selector.matchLabels.\"machine.openshift.io/cluster-api-machineset\" = .metadata.name
+| .spec.template.metadata.labels.\"machine.openshift.io/cluster-api-machineset\" = .metadata.name" \
+${dir}/openshift/99_openshift-cluster-api_worker-machineset-0.yaml > ${dir}/openshift/99_openshift-cluster-api_worker-machineset-additional.yaml
+
+yq-v4 eval ".spec.template.spec.providerSpec.value.ami.id = \"${amiid_workers_additional}\"
+            | .spec.template.spec.providerSpec.value.instanceType = \"${ADDITIONAL_WORKER_VM_TYPE}\"
+            " -i ${dir}/openshift/99_openshift-cluster-api_worker-machineset-additional.yaml
+
+oc create -f ${dir}/openshift/99_openshift-cluster-api_worker-machineset-additional.yaml &
```

Comment:
for debugging purposes, you may want a:

```sh
cat "${dir}/openshift/99_openshift-cluster-api_worker-machineset-additional.yaml"
```

here.

Or use [`${ARTIFACT_DIR}`][1] as your scratch space, instead of `${dir}/openshift`?

[1]: https://docs.ci.openshift.org/docs/architecture/step-registry/#available-environment-variables

Reply:
done